File: zig-cli/src/entity/note.zig
//! Note entity types.

const std = @import("std");

/// Represents a stored note.
pub const Note = struct {
    id: []const u8,
    title: []const u8,
    content: []const u8,
    created_at: f64,
    updated_at: f64,

    /// Returns a short ID for display (max 12 chars).
    pub fn shortId(self: Note) []const u8 {
        return if (self.id.len > 12) self.id[0..12] else self.id;
    }

    /// Returns a content preview (max 40 chars).
    pub fn contentPreview(self: Note) []const u8 {
        return if (self.content.len > 40) self.content[0..40] else self.content;
    }

    /// Returns formatted timestamp (simplified).
    pub fn formattedCreated(self: Note, buf: []u8) []const u8 {
        const timestamp_ms: i64 = @intFromFloat(self.created_at);
        const epoch_seconds: u64 = @intCast(@divTrunc(timestamp_ms, 1000));
        // Simple formatting - just show seconds since epoch for now
        return std.fmt.bufPrint(buf, "{d}", .{epoch_seconds}) catch "?";
    }
};

/// Data for creating a note.
pub const CreateNote = struct {
    title: []const u8,
    content: []const u8,
};

/// Data for updating a note.
pub const UpdateNote = struct {
    id: []const u8,
    title: []const u8,
    content: []const u8,
};

File: go-cli/internal/entity/note.go
// Package entity contains domain types.
package entity

import "time"

// Note represents a stored note.
type Note struct {
	ID        string  `json:"_id"`
	Title     string  `json:"title"`
	Content   string  `json:"content"`
	CreatedAt float64 `json:"createdAt"`
	UpdatedAt float64 `json:"updatedAt"`
}

// ShortID returns a truncated ID for display.
func (n Note) ShortID() string {
	if len(n.ID) > 12 {
		return n.ID[:12]
	}
	return n.ID
}

// ContentPreview returns a truncated content preview.
func (n Note) ContentPreview() string {
	if len(n.Content) > 40 {
		return n.Content[:40] + "..."
	}
	return n.Content
}

// FormattedCreated returns a human-readable creation time.
func (n Note) FormattedCreated() string {
	t := time.UnixMilli(int64(n.CreatedAt))
	return t.Format("2006-01-02 15:04")
}

// CreateNote contains data for creating a note.
type CreateNote struct {
	Title   string `json:"title"`
	Content string `json:"content"`
}

// UpdateNote contains data for updating a note.
type UpdateNote struct {
	ID      string `json:"id"`
	Title   string `json:"title"`
	Content string `json:"content"`
}
